pkgs/development/interpreters/jruby/default.nix

```diff
@@ -1,67 +1,106 @@
-{ lib, stdenv, callPackage, fetchurl, makeWrapper, jre }:
+{ lib, stdenv, callPackage, fetchFromGitHub, makeWrapper, jre, maven, ant, jrubyVersion ? "9.2.16.0" }:
 
 let
-# The version number here is whatever is reported by the RUBY_VERSION string
-rubyVersion = callPackage ../ruby/ruby-version.nix {} "2" "5" "7" "";
-jruby = stdenv.mkDerivation rec {
-  pname = "jruby";
-
-  version = "9.2.16.0";
+  # The version number here is whatever is reported by the RUBY_VERSION string
+  rubyVersion = callPackage ../ruby/ruby-version.nix { } "2" "5" "7" "";
+  jruby-src = fetchFromGitHub {
+    owner = "jruby";
+    repo = "jruby";
+    rev = "${jrubyVersion}";
+    sha256 = "14zs1xr9rb7ydna7kfr2xvsvgwpm8lr5ngmb8dqgdv1hq687h6wz";
+  };
+  # Nixpkgs works by disallowing Internet access in order to enforce
+  # reproducibility. One way around that however is if your build is already
+  # binary reproducible, which you can toggle by specifying outputHash.
+  # We create a maven repository (~/.m2) for use in the build further down.
+  maven-repository = stdenv.mkDerivation {
```

Comment:
Seeking input on how to make this easy to override in overlays, when there's a fixed-output-derivation involved.